tests/integration/test_run_app_from_bundle.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import subprocess
+import tarfile
+
+import utils
+
+
+def test_run_app_from_bundle(test_env, default_request, tmpdir):
+    """
+    Check that downloaded bundle could be used to run the application.
+
+    Process:
+    * Send new request to Cachito API
+    * Download a bundle from the request
+    * Run go build
+    * Run the application
+
+    Checks:
+    * Check that the state of request is complete
+    * Check that the bundle is properly downloaded
+    * Check that the application runs successfully
+    """
+    response = default_request.complete_response
+    assert response.status == 200
+    assert response.data["state"] == "complete"
+
+    file_name_tar = tmpdir.join(f"download_{str(response.id)}.tar.gz")
+    file_name = tmpdir.join(f"download_{str(response.id)}")
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    resp = client.download_bundle(response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(file_name)
+
+    app_name = test_env["run_app"]["app_name"]
+    subprocess.run(
+        ["go", "build", "-o", f"{file_name}/app/bin/{app_name}", f"{file_name}/app/main.go"],
```

Comment:
Would this work so that you don't assume a Linux/Unix style path?
```suggestion
        ["go", "build", "-o", str(file_name.join("app", "bin", app_name)), str(file_name.join("app", "main.go"))],
```